### PR TITLE
Add `full and wide` align controls to Query and Query Pagination blocks

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Related: https://github.com/WordPress/gutenberg/issues/27589

This PR adds `full and wide` align support to `Query/QueryLoop` and `Query Pagination` blocks.

Ideally we would want to handle this with a mechanism for `absorbing` the toolbar of nested block wrappers as described in this issue: https://github.com/WordPress/gutenberg/issues/7694.

That's why I haven't used the `supports.align` in `QueryLoop` and created an adhoc toolbar control for `align` in `Query` block. This is a UI choice for better user experience.

## How has this been tested?
Manually by adding `Query` and `Query Pagination` blocks and try different combinations of `align` options.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/query-loop/edit.js
Comment (youknowriad) [inline packages/block-library/src/query-loop/edit.js:138]: why d we need this? Isn't it done automatically in the hooks?

File: packages/block-library/src/query/block.json
Comment (youknowriad) [inline packages/block-library/src/query/block.json:45]: I don't really understand why the "query" block supports these flags? This is not a block that has any markup?

File: packages/block-library/src/query-loop/index.php
Comment (youknowriad) [inline packages/block-library/src/query-loop/index.php:50]: Isn't this something that can be applied automatically by the the support flag?

File: packages/block-library/src/query/block.json
Comment (ntsekouras) [reply packages/block-library/src/query/block.json:45]: Check my comment about the attributes. This is what we had discussed about `layout` as well, so `align` goes to `Query` now.

File: packages/block-library/src/query-loop/index.php
Comment (ntsekouras) [reply packages/block-library/src/query-loop/index.php:50]: Check my comment about the attributes. This is what we had discussed about layout as well, so align goes to Query now.

File: packages/block-library/src/query/block.json
Comment (mtias) [reply packages/block-library/src/query/block.json:45]: If we want to avoid all "layout" props from being present in the query we need something in the vein of #7694 for the outer wrapper

**Comment by mtias** [comment]
Agreed with the followup comment — we'd need to figure out what absorb in toolbar and passthrough mean in terms inner block properties. Making a wrapper be invisible or partially invisible (in the context of not affecting layout) is one of the oldest issues in https://github.com/WordPress/gutenberg/issues/7694

**Review by carolinan — APPROVED** [review]
I have nothing to add to the code review.
The PR solves the problem described in the issue. 
Full width and wide blocks inside a query loop appear with the correct widths inside the editor.

**Review by aristath — APPROVED** [review]
This looks OK to me, it just needs to be rebased before merging 👍

**Comment by ntsekouras** [comment]
Thanks for reviewing!

>This looks OK to me, it just needs to be rebased before merging 👍

Besides rebase and a small change to the selectors (not use `getBlocks`), I have this on hold due to @youknowriad 's PR about alignments: https://github.com/WordPress/gutenberg/pull/29335.

**Comment by paaljoachim** [comment]
> Besides rebase and a small change to the selectors (not use getBlocks), I have this on hold due to @youknowriad 's PR about alignments: #29335.

PR has been merged so I assume this PR can be rebased and adjusted and then merged.

**Comment by ntsekouras** [comment]
>PR has been merged so I assume this PR can be rebased and adjusted and then merged.

Yes, it must be in combination with adding a wrapper for Query block.

**Comment by ntsekouras** [comment]
Closing in favor of: https://github.com/WordPress/gutenberg/pull/30804

**Pull request closed without merging**
